src/creeps/Zombie.cpp

```diff
@@ -153,72 +276,87 @@ void Zombie::generateMove() {
 
     // reset the node maps
-    memset(closedNodes, 0, sizeof(closedNodes[0][0]) * ROW * COL);
-    memset(openNodes, 0, sizeof(openNodes[0][0]) * ROW * COL);
+    memset(closedNodes, 0, sizeof(int) * ROWS * COLS);
+    memset(openNodes, 0, sizeof(int) * ROWS * COLS);
+    memset(dirMap, 0, sizeof(int) * ROWS * COLS);
 
-    const int xNodeStart = static_cast<int>(xStart / TILE_SIZE);
-    const int yNodeStart = static_cast<int>(yStart / TILE_SIZE);
-    const int xNodeDest = static_cast<int>(xDest / TILE_SIZE);
-    const int yNodeDest = static_cast<int>(yDest / TILE_SIZE);
+    int xNodeStart = static_cast<int> (start.second + TILE_OFFSET) / TILE_SIZE;
+    int yNodeStart = static_cast<int> (start.first + TILE_OFFSET) / TILE_SIZE;
+    int xNodeDest = static_cast<int> (dest.second + TILE_OFFSET) / TILE_SIZE - 1;
+    int yNodeDest = static_cast<int> (dest.first + TILE_OFFSET) / TILE_SIZE - 1;
 
-    // create the start node and push the start node into open list
-    Node curNode(xNodeStart, yNodeStart);
+    // create the start node and push into open list
+    curNode = Node(xNodeStart, yNodeStart, 0, 0);
     curNode.updatePriority(xNodeDest, yNodeDest);
     pq[index].push(curNode);
 
     // A* path finding
     while (!pq[index].empty()) {
         // get the current node with the highest priority from open list
-        curNode = pq[index].top();
+        curNode = Node(pq[index].top().getXPos(), pq[index].top().getYPos(),
+                       pq[index].top().getLevel(), pq[index].top().getPriority());
 
-        x = curNode.getXPos();
-        y = curNode.getYPos();
+        curRow = curNode.getXPos();
+        curCol = curNode.getYPos();
 
         // remove the node from the open list
         pq[index].pop();
-        openNodes[x][y] = 0;
 
-        // mark it on the closed nodes map
-        closedNodes[x][y] = 1;
+        // mark it on open/close map
+        openNodes[curRow][curCol] = 0;
+        closedNodes[curRow][curCol] = 1;
 
         // quit searching when the destination is reached
-        if (x == xNodeDest && y == yNodeDest) {
+        if (curRow == xNodeDest && curCol == yNodeDest) {
             // generate the path from destination to start
             // by following the directions
             path = "";
-            while (!(x == xNodeStart && y == yNodeStart)) {
-                j = dirMap[x][y];
-                path = static_cast<char>('0' + (j + DIR_CAP / 2) % DIR_CAP) + path;
-                x += MX[j];
-                y += MY[j];
+            while (!(curRow == xNodeStart && curCol == yNodeStart)) {
+                j = dirMap[curRow][curCol];
+                c = '0' + (j + DIR_CAP/2)%DIR_CAP;
+                path = c + path;
+                curRow += MY[j];
+                curCol += MX[j];
             }
 
             // empty the leftover nodes
-            pq[index] = priority_queue<Node>();
+            while (!pq[index].empty()) {
+                 pq[index].pop();
+            }
```

Comment:
Why change this?  They both do the same thing but `pq[index] = priority_queue<Node>()` was a quicker and cleaner solution.